src/upgrader.ts

```diff
@@ -258,39 +258,51 @@ export class DefaultUpgrader extends EventEmitter<UpgraderEvents> implements Upg
 
     log('Starting the outbound connection upgrade')
 
+    // If the transport natively supports encryption, skip connection
+    // protector and encryption
+
     // Protect
     let protectedConn = maConn
-    const protector = this.components.getConnectionProtector()
+    if ((opts?.muxerFactory) == null) {
```

Comment:
After the [suggested change](https://github.com/libp2p/js-libp2p-interfaces/pull/290#discussion_r988230911) in the interface repo we can make this decision on explicit settings rather than the lack of presence of an overriding muxer factory:

```suggestion
    if (opts.skipProtection !== true) {
```